File: practice_problems/arrays/js/contiguousSum.js
'use strict';
'esversion: 6';

const conSum = (arrNums) => {
  if (arrNums.length < 1) return 0;
  if (arrNums.length == 1) return arrNums[0];

  let currentSum = 0;
  let maxSum = arrNums[0];

  arrNums.forEach( (current) => {
    currentSum = currentSum + current >= current ? currentSum + current : current;
    maxSum = maxSum >= currentSum ? maxSum : currentSum;
  });

  return maxSum;
};
let foo = [4, -2, 3, -10, 1, 2, 3];

console.log(conSum(foo));

File: practice_problems/arrays/python/contiguous_sum.py
def contiguous_sum(arr):
    current_sum = 0
    current_max = arr[0]

    for current in arr:
        if current + current_sum >= current_sum:
            current_sum = current + current_sum
        else:
            current_sum = current

        if current_sum >= current_max:
            current_max = current_sum

    return current_max


print(contiguous_sum([1,2,3]))
